### PR TITLE
[ISSUE #3593] entrySet() replace keySet()

### PR DESCRIPTION
**Pull request by zhaohai1299002788**
There are several inefficient iterations of Map, in such code, keySet() is retrieved from Map firstly, and values are got in each iteration by key. In my opinion, entrySet() should be used, so that keys and values can be fetched in one time.

### CODE REVIEW

**Review by panzhi33 — APPROVED** [review]

**Review by guyinyou — APPROVED** [review]

File: test/src/main/java/org/apache/rocketmq/test/util/MQAdmin.java [hunk 1]
```diff
@@ -123,10 +124,10 @@ public static boolean isBrokerExist(String ns, String ip) {
             return false;
         } else {
             HashMap<String, BrokerData> brokers = clusterInfo.getBrokerAddrTable();
-            for (String brokerName : brokers.keySet()) {
-                HashMap<Long, String> brokerIps = brokers.get(brokerName).getBrokerAddrs();
-                for (long brokerId : brokerIps.keySet()) {
-                    if (brokerIps.get(brokerId).contains(ip))
+            for (Entry<String, BrokerData> brokerNameEntry : brokers.entrySet()) {
```
Comment (yuz10) [inline test/src/main/java/org/apache/rocketmq/test/util/MQAdmin.java:127]: brokerNameEntry could be renamed to brokerEntry

File: tools/src/main/java/org/apache/rocketmq/tools/command/CommandUtil.java [hunk 2]
```diff
@@ -63,12 +64,12 @@
             String masterAddr = brokerData.getBrokerAddrs().get(MixAll.MASTER_ID);
             masterAndSlaveMap.put(masterAddr, new ArrayList<String>());
 
-            for (Long id : brokerData.getBrokerAddrs().keySet()) {
-                if (brokerData.getBrokerAddrs().get(id) == null || id == MixAll.MASTER_ID) {
+            for (Entry<Long, String> idEntry : brokerData.getBrokerAddrs().entrySet()) {
+                if (idEntry.getValue() == null || idEntry.getKey() == MixAll.MASTER_ID) {
```
Comment (yuz10) [inline tools/src/main/java/org/apache/rocketmq/tools/command/CommandUtil.java:68]: idEntry could be renamed to brokerAddrEntry

File: tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetBrokerConfigCommand.java [hunk 3]
```diff
@@ -83,19 +86,19 @@ public void execute(final CommandLine commandLine, final Options options,
                 Map<String, List<String>> masterAndSlaveMap
                     = CommandUtil.fetchMasterAndSlaveDistinguish(defaultMQAdminExt, clusterName);
 
-                for (String masterAddr : masterAndSlaveMap.keySet()) {
+                for (Entry<String, List<String>> masterAddrEntry : masterAndSlaveMap.entrySet()) {
```
Comment (yuz10) [inline tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetBrokerConfigCommand.java:89]: masterAddrEntry could be renamed to masterAndSlaveEntry

File: tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetBrokerConfigCommand.java [hunk 4]
```diff
@@ -121,8 +124,8 @@ protected void getAndPrint(final MQAdminExt defaultMQAdminExt, final String prin
             return;
         }
 
-        for (Object key : properties.keySet()) {
-            System.out.printf("%-50s=  %s\n", key, properties.get(key));
+        for (Entry<Object, Object> keyEntry : properties.entrySet()) {
+            System.out.printf("%-50s=  %s\n", keyEntry.getKey(), keyEntry.getValue());
```
Comment (yuz10) [inline tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetBrokerConfigCommand.java:128]: keyEntry could be renamed to entry, and so is the obove

File: tools/src/main/java/org/apache/rocketmq/tools/command/consumer/GetConsumerConfigSubCommand.java [hunk 5]
```diff
@@ -63,14 +65,14 @@ public void execute(CommandLine commandLine, Options options,
             List<ConsumerConfigInfo> consumerConfigInfoList = new ArrayList<>();
             ClusterInfo clusterInfo = adminExt.examineBrokerClusterInfo();
             Map<String, Set<String>> clusterAddrTable = clusterInfo.getClusterAddrTable();
-            for (String brokerName : clusterInfo.getBrokerAddrTable().keySet()) {
-                String clusterName = this.getClusterName(brokerName, clusterAddrTable);
-                String brokerAddress = clusterInfo.getBrokerAddrTable().get(brokerName).selectBrokerAddr();
+            for (Entry<String, BrokerData> brokerNameEntry : clusterInfo.getBrokerAddrTable().entrySet()) {
```
Comment (yuz10) [inline tools/src/main/java/org/apache/rocketmq/tools/command/consumer/GetConsumerConfigSubCommand.java:68]: brokerNameEntry could be rename to brokerEntry

File: tools/src/main/java/org/apache/rocketmq/tools/command/export/ExportConfigsCommand.java [hunk 6]
```diff
@@ -79,10 +81,10 @@ public void execute(CommandLine commandLine, Options options, RPCHook rpcHook)
             Map<String, Properties> brokerConfigs = new HashMap<>();
             Map<String, List<String>> masterAndSlaveMap
                 = CommandUtil.fetchMasterAndSlaveDistinguish(defaultMQAdminExt, clusterName);
-            for (String masterAddr : masterAndSlaveMap.keySet()) {
-                Properties masterProperties = defaultMQAdminExt.getBrokerConfig(masterAddr);
+            for (Entry<String, List<String>> masterAddrEntry : masterAndSlaveMap.entrySet()) {
```
Comment (yuz10) [inline tools/src/main/java/org/apache/rocketmq/tools/command/export/ExportConfigsCommand.java:84]: masterAddrEntry could be renamed to masterAndSlaveEntry

File: tools/src/main/java/org/apache/rocketmq/tools/command/namesrv/GetNamesrvConfigCommand.java [hunk 7]
```diff
@@ -66,11 +69,11 @@ public void execute(final CommandLine commandLine, final Options options,
 
             Map<String, Properties> nameServerConfigs = defaultMQAdminExt.getNameServerConfig(serverList);
 
-            for (String server : nameServerConfigs.keySet()) {
+            for (Entry<String, Properties> serverEntry : nameServerConfigs.entrySet()) {
```
Comment (yuz10) [inline tools/src/main/java/org/apache/rocketmq/tools/command/namesrv/GetNamesrvConfigCommand.java:72]: serverEntry could be renamed to nameServerConfigEntry

**Review by yuz10 — CHANGES_REQUESTED** [review]

File: test/src/main/java/org/apache/rocketmq/test/util/MQAdmin.java (hunk 1 above)
Comment (zhaohai1299002788) [reply test/src/main/java/org/apache/rocketmq/test/util/MQAdmin.java]: already edited

File: tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetBrokerConfigCommand.java (hunk 3 above)
Comment (zhaohai1299002788) [reply tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetBrokerConfigCommand.java]: already edited

File: tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetBrokerConfigCommand.java (hunk 4 above)
Comment (zhaohai1299002788) [reply tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetBrokerConfigCommand.java]: already edited

File: tools/src/main/java/org/apache/rocketmq/tools/command/consumer/GetConsumerConfigSubCommand.java (hunk 5 above)
Comment (zhaohai1299002788) [reply tools/src/main/java/org/apache/rocketmq/tools/command/consumer/GetConsumerConfigSubCommand.java]: already edited

File: tools/src/main/java/org/apache/rocketmq/tools/command/export/ExportConfigsCommand.java (hunk 6 above)
Comment (zhaohai1299002788) [reply tools/src/main/java/org/apache/rocketmq/tools/command/export/ExportConfigsCommand.java]: already edited

File: tools/src/main/java/org/apache/rocketmq/tools/command/namesrv/GetNamesrvConfigCommand.java (hunk 7 above)
Comment (zhaohai1299002788) [reply tools/src/main/java/org/apache/rocketmq/tools/command/namesrv/GetNamesrvConfigCommand.java]: already edited

**Review by yuz10 — APPROVED** [review]

**Review by Git-Yang — APPROVED** [review]

**Comment by Git-Yang** [comment]
/home/travis/build/apache/rocketmq/client/src/main/java/org/apache/rocketmq/client/consumer/rebalance/AllocateMachineRoomNearby.java:23:8: error: Unused import - java.util.Set.

**Review by RongtongJin — APPROVED** [review]

**Review by Git-Yang — APPROVED** [review]

**Comment by Git-Yang** [comment]
link https://github.com/apache/rocketmq/issues/3593

**Pull request merged**